### PR TITLE
Implement `std::pair` and `std::array`

### PR DESCRIPTION
**Pull request by no92**
This is needed by the new bragi additions for managarm/mlibc.

### CODE REVIEW

**Review by ElectrodeYT — APPROVED** [review]
LGTM

File: stage2/include/array [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	[[nodiscard]] element &operator[](std::size_t n) noexcept {
+		assert(n < this->size());
+		return elements_[n];
+	}
+
+	[[nodiscard]] pointer data() noexcept {
+		return static_cast<pointer>(elements_);
+	}
+
+	[[nodiscard]] pointer data() const noexcept {
```
Comment (Qwinci) [inline stage2/include/array:46]: This should return a const_pointer (which is a typedef to const element *).

File: stage2/include/array [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+		return static_cast<pointer>(elements_);
+	}
+
+	[[nodiscard]] iterator begin() noexcept {
+		return iterator(data());
+	}
+
+	[[nodiscard]] iterator begin() const noexcept {
```
Comment (Qwinci) [inline stage2/include/array:54]: Same here, except const_iterator.

File: stage2/include/array [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		return iterator(data());
+	}
+
+	[[nodiscard]] iterator end() noexcept {
+		return iterator(data() + count);
+	}
+
+	[[nodiscard]] iterator end() const noexcept {
```
Comment (Qwinci) [inline stage2/include/array:62]: Same here.

File: stage2/include/array [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	typedef element *pointer;
+	typedef element *iterator;
+
+	typename __array_container<element, count>::type elements_;
+
+	constexpr std::size_t size() const noexcept {
+		return count;
+	}
+
+	[[nodiscard]] element &operator[](std::size_t n) noexcept {
```
Comment (Qwinci) [inline stage2/include/array:37]: Might be nice to add the const ref variant of this too.

**Review by Qwinci — CHANGES_REQUESTED** [review]
Looks fine to me (not taking the missing std functions into account) other than a few const related issues.

**Comment by no92** [comment]
Addressed the comments raised, and added implementations of `max_size()` and `empty()`.

**Review by Qwinci — APPROVED** [review]

File: stage2/include/array [hunk 5]
```diff
@@ -0,0 +1,86 @@
+#ifndef _CXXSHIM_ARRAY
+#define _CXXSHIM_ARRAY
+
+#include <cstddef>
+
+namespace std {
+
+template<typename element, std::size_t count>
```
Comment (ArsenArsen) [inline stage2/include/array:8]: these should be UpperCamelCase

Comment (no92) [reply stage2/include/array:8]: See #20.

**Pull request merged**